core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerFacade.java

```diff
@@ -66,14 +70,69 @@ boolean isPlaying() {
         return exoPlayer != null && exoPlayer.getPlayWhenReady();
     }
 
+    boolean isPlayingAdvert() {
+        return isPlaying() && exoPlayer.isPlayingAd();
+    }
+
+    boolean isPlayingContent() {
+        return isPlaying() && !exoPlayer.isPlayingAd();
+    }
+
+    long mediaDurationInMillis() throws IllegalStateException {
+        assertVideoLoaded();
+        return exoPlayer.getDuration();
+    }
+
     long playheadPositionInMillis() throws IllegalStateException {
         assertVideoLoaded();
         return exoPlayer.getCurrentPosition();
     }
 
-    long mediaDurationInMillis() throws IllegalStateException {
+    long advertBreakDurationInMillis() throws IllegalStateException {
         assertVideoLoaded();
-        return exoPlayer.getDuration();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int currentAdGroupIndex = exoPlayer.getCurrentAdGroupIndex();
+        int advertCount = period.getAdCountInAdGroup(currentAdGroupIndex);
+
+        long advertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(advertBreakDurationInMicros);
+    }
+
+    long positionInAdvertBreakInMillis() throws IllegalStateException {
+        assertVideoLoaded();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int advertCount = exoPlayer.getCurrentAdIndexInAdGroup();
+
+        long playedAdvertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(playedAdvertBreakDurationInMicros) + playheadPositionInMillis();
+    }
+
+    private long combinedAdvertDurationInGroup(Timeline.Period period, int numberOfAdvertsToInclude) {
+        int adGroupIndex = exoPlayer.getCurrentAdGroupIndex();
+        long advertBreakDurationInMicros = 0;
+        NoPlayerAdsLoader noPlayerAdsLoader = adsLoader.get();
+
+        for (int i = 0; i < numberOfAdvertsToInclude; i++) {
+            long advertDurationInMicros = period.getAdDurationUs(adGroupIndex, i);
+            if (advertDurationInMicros == C.TIME_UNSET) {
+                advertDurationInMicros = noPlayerAdsLoader.advertDurationBy(adGroupIndex, i);
+            }
+            advertBreakDurationInMicros += advertDurationInMicros;
```

Comment:
We can split this into two parts with an `if-else` and avoid reassignment:
```java
if (advertDurationInMicros == C.TIME_UNSET) {
  advertBreakDurationInMicros += noPlayerAdsLoader.advertDurationBy(adGroupIndex, i);
} else {
  advertBreakDurationInMicros += period.getAdDurationUs(adGroupIndex, i);
}
```

Reply:
Am I missing something? We'd need to do two calls to `period.getAdDurationUs`, which is why we have the reassignment.

Reply:
Ah! true, my bad, I thought the condition was not based on that value 🤐